KubeArmor/feeder/policyMatcher.go

```diff
@@ -12,450 +13,348 @@ import (
 
-								match.Action = path.Action
+	return match
+}
 
-								matches.Policies = append(matches.Policies, match)
-							} else if len(src.Directory) > 0 {
-								match := tp.MatchPolicy{}
+func newProcDirMatchPolicy(policyName, source string, dir tp.ProcessDirectoryType) tp.MatchPolicy {
+	match := tp.MatchPolicy{
+		PolicyName: policyName,
+		Severity:   strconv.Itoa(dir.Severity),
+		Tags:       dir.Tags,
+		Message:    dir.Message,
+		Source:     source,
+		Operation:  "Process",
+		Resource:   dir.Directory,
+		Action:     dir.Action,
+	}
 
-								match.PolicyName = secPolicy.Metadata["policyName"]
+	return match
+}
 
-								match.Severity = strconv.Itoa(path.Severity)
-								match.Tags = path.Tags
-								match.Message = path.Message
+func newFilePathMatchPolicy(policyName, source string, path tp.FilePathType) tp.MatchPolicy {
+	match := tp.MatchPolicy{
+		PolicyName: policyName,
+		Severity:   strconv.Itoa(path.Severity),
+		Tags:       path.Tags,
+		Message:    path.Message,
+		Source:     source,
+		Operation:  "File",
+		Resource:   path.Path,
+		Action:     path.Action,
+	}
 
-								match.Source = src.Directory
-								match.Operation = "Process"
-								match.Resource = path.Path
+	return match
+}
 
-								match.Action = path.Action
+func newFileDirMatchPolicy(policyName, source string, dir tp.FileDirectoryType) tp.MatchPolicy {
+	match := tp.MatchPolicy{
+		PolicyName: policyName,
+		Severity:   strconv.Itoa(dir.Severity),
+		Tags:       dir.Tags,
+		Message:    dir.Message,
+		Source:     source,
+		Operation:  "File",
+		Resource:   dir.Directory,
+		Action:     dir.Action,
+	}
 
-								matches.Policies = append(matches.Policies, match)
-							}
-						}
-					}
-				}
-			}
+	return match
+}
 
-			if len(secPolicy.Spec.Process.MatchDirectories) > 0 {
-				for _, dir := range secPolicy.Spec.Process.MatchDirectories {
-					if len(dir.FromSource) == 0 {
-						match := tp.MatchPolicy{}
+func newNetProtMatchPolicy(policyName, source, resource string, netProt tp.NetworkProtocolType) tp.MatchPolicy {
+	match := tp.MatchPolicy{
+		PolicyName: policyName,
+		Severity:   strconv.Itoa(netProt.Severity),
+		Tags:       netProt.Tags,
+		Message:    netProt.Message,
+		Source:     source,
+		Operation:  "Network",
+		Resource:   resource,
+		Action:     netProt.Action,
+	}
 
-						match.PolicyName = secPolicy.Metadata["policyName"]
+	return match
+}
 
-						match.Severity = strconv.Itoa(dir.Severity)
-						match.Tags = dir.Tags
-						match.Message = dir.Message
+func getProtocolFromName(proto string) string {
+	switch strings.ToLower(proto) {
+	case "tcp":
+		return "type=SOCK_STREAM"
+	case "udp":
+		return "type=SOCK_DGRAM"
+	case "icmp":
+		return "type=SOCK_RAW protocol=1"
+	default:
+		return ""
+	}
+}
 
-						match.Source = ""
-						match.Operation = "Process"
-						match.Resource = dir.Directory
+func newCapMatchPolicy(policyName, source, resource string, cap tp.CapabilitiesCapabilityType) tp.MatchPolicy {
```

Comment:
Here, net_raw capability is for networking; thus, the operation was network.
However, there are other capabilities. Please make room for them.

Reply:
Ok. I'm on it. Thanks.